personio-recruiting-api.yaml

```diff
@@ -31,183 +21,472 @@ paths:
+        external_posting_id:
+          type: string
+          description: |-
+            The external id of this job posting.
+
+            See https://mycompany.personio.de/configuration/recruiting/channels
+        message:
+          type: string
+          description: The applicant supplied free-text message.
+        tags:
+          type: array
+          description: |-
+            Tags to be associated with this application. Non-existing tags will be created.
+
+            See https://mycompany.personio.de/configuration/recruiting/tags
+          items:
+            type: string
+        files:
+          description: |-
+            References to previously uploaded files. Required categories depend on your personio settings.
+
+            See also 'uploaded_document' model.
+          type: array
+          items:
+            type: object
+            properties:
+              uuid:
+                type: string
+                format: uuid
+                description: Reference to a previously uploaded file. Use the `uuid` value returned from the documents endpoint here.
+              original_filename:
+                type: string
+              category:
+                type: string
+                description: Possible and required values depend on your personio settings.
+            required:
+              - uuid
+              - original_filename
+              - category
+        attributes:
+          type: array
+          description: |-
+            Attributes of this applications.
+            Required attributes depend on your personio settings, so do the format or possible values.
+
+            See https://mycompany.personio.de/configuration/recruiting/attributes
+          items:
+            type: object
+            properties:
+              id:
+                type: string
+                description: 'id of the attribute, found in your personio settings.'
+              value:
+                type: string
+                description: |-
+                  Value of the attribute.
+                  For OPTION attributes, this must be one of the predefined options.
+                  For DATE attributes, this needs to follow ISO 8601 Local date, i.e. `2021-04-30`
+            required:
+              - id
+              - value
+      required:
+        - first_name
+        - last_name
+        - email
+        - job_position_id
+        - tags
+        - files
+        - attributes
+    uploaded_document:
+      type: object
+      properties:
+        uuid:
+          type: string
+          format: uuid
+          description: The UUID of this file. Can be attached to an application by including this uuid in that request.
+        size:
+          type: integer
+          description: Uploaded file size in bytes.
+        mimetype:
+          type: string
+          description: Detected Mime Type of this file.
+        original_filename:
```

Comment:
This is actually `originalFilename`. I tested it.

Reply:
leaving this as it is since we changed it to snake case now in the service